### PR TITLE
Fix permissions for the social auth user

### PR DESCRIPTION
**Pull request by athineouc**

### CODE REVIEW

File: wayneapp/services/permission_manager.py [hunk 1]
```diff
@@ -0,0 +1,35 @@
+from django.contrib.auth.models import AnonymousUser, Permission
+from rest_framework.request import Request
+from wayneapp.constants import ControllerConstants
+
+
+class PermissionManager:
+
+    def has_save_permission(self, business_entity: str, request: Request) -> bool:
```
Comment (lleithold) [inline wayneapp/services/permission_manager.py]: Why don't you use staticmethods here?

File: wayneapp/authentication/social_authentication.py [hunk 2]
```diff
@@ -1,10 +1,16 @@
...
 
+from wayneapp.services import SchemaLoader
+from wayneapp.services.permission_manager import PermissionManager
+
 
 def process_roles(details, user, **kwargs):
-    if User.objects.filter(username=user.get_username).exists() is False:
+    schema_loader = SchemaLoader()
+    permission_manager = PermissionManager()
+    if not User.objects.filter(username=user.get_username).exists():
```
Comment (lleithold) [inline wayneapp/authentication/social_authentication.py]: So this is executed only once when the new user is created right? But what if any new business entities are added after the user was created?

File: wayne/settings.py [hunk 3]
```diff
@@ -213,8 +213,6 @@
 WAYNE_ADMIN = {
     'JS_URL': 'js/wayne-admin.js',
     'CSS_URL': 'css/wayne-admin.css',
-    # feature flag to enable delete functionality in the admin tool
-    'DELETE_ENABLED': True
```
Comment (lleithold) [inline wayne/settings.py]: Why did you remove this feature flag?

File: wayneapp/authentication/social_authentication.py (hunk 2 above)
Comment (tobi812) [inline wayneapp/authentication/social_authentication.py]: I have a suggestion here: Could we simply attach a group to the user, that contains all permission?

File: wayneapp/services/permission_manager.py (hunk 1 above)
Comment (athineouc) [reply wayneapp/services/permission_manager.py]: Actually i did it like this at first but then I just thought that it would be nice to not have everything static. I am fine also to have staticmethods here.

File: wayneapp/authentication/social_authentication.py (hunk 2 above)
Comment (athineouc) [reply wayneapp/authentication/social_authentication.py]: You are right. I did not think this. Should i add a view permission to all existing users when we create the model in generatemodels command?

Comment (lleithold) [reply wayneapp/authentication/social_authentication.py]: Yes, but when we have a permission group, we only have to update the group, not every user.

File: wayne/settings.py (hunk 3 above)
Comment (athineouc) [reply wayne/settings.py]: At first i also delete the has_delete_permission function we had in the admin.py file. I did that because i want the default implementation of this function from django where it checks if the user has the permission to delete. Then i deleted also this variable from the settings because i saw that we just used it to enable or disable the delete operations for all the users. Please correct me if i miss something and we also use this settings somewhere else.

File: wayneapp/admin.py [hunk 4]
```diff
@@ -2,31 +2,33 @@
...
     list_display = ['key', 'version']
     search_fields = ['key', 'version']
 
     _entity_manager = BusinessEntityManager()
 
+    def has_view_permission(self, request, obj=None):
+        if super().has_view_permission(request, obj):
+            return True
+
+        return request.user.groups\
```
Comment (tobi812) [inline wayneapp/admin.py]: Shouldn't we check against a permission here and not a group?

Comment (athineouc) [reply wayneapp/admin.py]: Yes we can also do it, it is the same result. Maybe also it will be more clear. But then do we need to create the group just for this one permission?

File: wayneapp/apps.py [hunk 5]
```diff
@@ -1,7 +1,25 @@
 from django.apps import AppConfig
+from django.db.models.signals import post_migrate
 from wayne import settings
 
+from wayneapp.constants import GroupConstants
+
+
+def create_group_for_view_permission(sender, **kwargs):
```
Comment (tobi812) [inline wayneapp/apps.py]: Can we move this to another file/class?
